lib/spack/spack/package_base.py

```diff
@@ -1037,8 +1039,46 @@ def resolve_binary_provenance(self) -> None:
         Base implementation will look up git commits when appropriate.
         Packages may override this implementation for custom implementations
         """
-        # TODO in follow on PR adding here so SNL team can begin work ahead of spack core
-        pass
+
+        # early return cases, don't overwrite user intention
+        # commit pre-assigned or develop specs don't need commits changed
+        # since this would create un-necessary churn
+        if "commit" in self.spec.variants or self.spec.is_develop:
+            return
+
+        sha = None
+
+        if is_git_version(str(self.spec.version)):
+            ref = self.spec.version.ref
+        else:
+            tag = self.version_or_package_attr("tag", self.spec.version, None)
+            branch = self.version_or_package_attr("branch", self.spec.version, None)
+            assert not (tag and branch)
+            ref = tag or branch
+
+        assert ref, "Missing git ref"
+
+        if not sha:
```

Comment:
I think this is guaranteed at this point, so we can avoid the conditional. It will also be more legible if you move the line initializing `sha` to here.